### PR TITLE
Feature/dfe 968 convert add edit data page

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
This PR:
- adds API endpoints to the "Release Setup" and "Edit Release Setup" pages, along with mock responses (https://eesadminprototype.z33.web.core.windows.net/prototypes/publication-create-new-absence-config) and https://eesadminprototype.z33.web.core.windows.net/prototypes/publication-create-new-absence-config-edit)
- adds the implementation of the "Data Uploads" tab on the "Manage Data" page (https://eesadminprototype.z33.web.core.windows.net/prototypes/publication-create-new-absence-data) but not currently the "File Uploads" tab, which'll be dealt with separately
- introduces a new common FormFieldFileSelector component for using file upload input fields with Formik (that was fun!)
- includes some cleanup and refactoring of common code
- includes some cleanup of Admin Dashboard components (removing little ones and placing them inline into others, and renaming a couple)

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx [hunk 1]
```diff
@@ -1,38 +1,273 @@
...
+                      fileInput.value = '';
+                    });
+                });
+            }}
+            validationSchema={Yup.object<FormValues>({
+              subjectTitle: Yup.string().required('Enter a subject title'),
+              dataFile: Yup.mixed().required('Choose a data file'),
+              metadataFile: Yup.mixed().required('Choose a metadata file'),
+            })}
+            render={(form: FormikProps<FormValues>) => {
```
Comment (richardpopplehive) [inline src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx]: I don't know enough about react - but why do we have a render method AND the html above?

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleasePublishStatusPage.tsx [hunk 2]
```diff
@@ -13,28 +11,9 @@ const ReleasePublishStatusPage = ({
...
-    setPublicationTitle(owningPublication ? owningPublication.title : '');
-  }, [releaseId]);
-
   return (
-    <ReleasePageTemplate
-      publicationTitle={publicationTitle}
-      releaseId={releaseId}
-    >
-      {release && <h2 className="govuk-heading-m">Set publish status</h2>}
+    <ReleasePageTemplate publicationTitle="TODO" releaseId={releaseId}>
```
Comment (richardpopplehive) [inline src/explore-education-statistics-admin/src/pages/edit-release/ReleasePublishStatusPage.tsx]: I guess we don't have the publicationTitle

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx [hunk 3]
```diff
@@ -61,11 +62,11 @@ const ReleaseSetupEditPage = ({
...
 
   useEffect(() => {
-    const release = DummyPublicationsData.getReleaseSetupDetails(releaseId);
-
-    setReleaseSetupDetails(release);
-    setTimePeriodCoverageGroups(DummyReferenceData.timePeriodCoverageGroups);
-    setReleaseTypes(DummyReferenceData.releaseTypeOptions);
+    service.getReleaseSetupDetails(releaseId).then(release => {
+      setReleaseSetupDetails(release);
+      setTimePeriodCoverageGroups(DummyReferenceData.timePeriodCoverageGroups);
```
Comment (richardpopplehive) [inline src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx]: FYI this call exists now for the next iteration - api/meta/timeidenitifiers - I think that's what you'd want

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx [hunk 4]
```diff
@@ -61,11 +62,11 @@ const ReleaseSetupEditPage = ({
...
+      setReleaseTypes(DummyReferenceData.releaseTypeOptions);
```
Comment (richardpopplehive) [inline src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx]: This call will also exist shortly (when the pull request is in) /api/meta/releasetypes

File: src/explore-education-statistics-admin/src/pages/admin-dashboard/components/AdminDashboardPublicationSummary.tsx [hunk 5]
```diff
@@ -1,7 +1,7 @@
-import { AdminDashboardPublication } from '@admin/services/api/dashboard/types';
+import DashboardReleaseSummary from '@admin/pages/admin-dashboard/components/DashboardReleaseSummary';
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/admin-dashboard/components/AdminDashboardPublicationSummary.tsx]: No, it's the default export from that file, so you don't need them :)

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseBuildTablesPage.tsx [hunk 6]
```diff
@@ -11,28 +9,9 @@ interface MatchProps {
...
-    setPublicationTitle(owningPublication ? owningPublication.title : '');
-  }, [releaseId]);
-
   return (
-    <ReleasePageTemplate
-      publicationTitle={publicationTitle}
-      releaseId={releaseId}
-    >
-      {release && <h2 className="govuk-heading-m">Build tables</h2>}
+    <ReleasePageTemplate publicationTitle="TODO" releaseId={releaseId}>
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseBuildTablesPage.tsx]: Not done the service calls for this component yet, but when I do I will wire this up as per the other sections worked on so far

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseContentPage.tsx [hunk 7]
```diff
@@ -11,28 +9,9 @@ interface MatchProps {
...
-    setPublicationTitle(owningPublication ? owningPublication.title : '');
-  }, [releaseId]);
-
   return (
-    <ReleasePageTemplate
-      publicationTitle={publicationTitle}
-      releaseId={releaseId}
-    >
-      {release && <h2 className="govuk-heading-m">Add / edit content</h2>}
+    <ReleasePageTemplate publicationTitle="TODO" releaseId={releaseId}>
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseContentPage.tsx]: Not done the service calls for this component yet, but when I do I will wire this up as per the other sections worked on so far

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx [hunk 8]
```diff
@@ -1,38 +1,273 @@
...
+                  metadataFile: values.metadataFile as File,
+                })
+                .then(_ => service.getReleaseDataFiles(releaseId))
+                .then(setDataFiles)
+                .then(_ => {
+                  actions.resetForm();
+                  document
+                    .querySelectorAll(`#${formId} input[type='file']`)
+                    .forEach(input => {
+                      const fileInput = input as HTMLInputElement;
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx]: Yeah - unfortunately Formik and input fields of type "file" don't play nice together, so we have to manage the form state and the input field state manually, and also clear out the input field values manually too.  This is due to browser security constraints that only allow you to programatically set the value of a file upload field to the empty string.

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx]: This is a special Formik method that renders just the Form that it controls.

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleasePublishStatusPage.tsx (hunk 2 above)
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleasePublishStatusPage.tsx]: Not done the service calls for this component yet, but when I do I will wire this up as per the other sections worked on so far

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx (hunk 3 above)
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx]: Cool thanks!

File: src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx (hunk 4 above)
Comment (duncan-at-hiveit) [reply src/explore-education-statistics-admin/src/pages/edit-release/ReleaseSetupEditPage.tsx]: Cool thanks!

**Pull request merged**